examples/TestDriver/src/pages/nav.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import { Button, StyleSheet, Text, View } from 'react-native';
+import { createStackNavigator } from 'react-navigation';
+
+import { makeSentinelButton } from '../sentinelUtilities';
+
+const BaseView = props => {
+  return (
+    <View style={styles.container}>
+      <Button
+        title="Navigate Stack"
+        testID="navigate_stack"
+        onPress={() => props.navigation.navigate('StackCard')}
+      />
+      <Button
+        title="Navigate Modal"
+        testID="navigate_modal"
+        onPress={() => props.navigation.navigate('ModalStack')}
+      />
+      {makeSentinelButton('Nav')}
```

Comment:
Ah I see they're tested by expecting navigation events. Something feels a little fishy about how this `makeSentinelButton` thing is different than the other things. What exactly is it meant to do?

Reply:
Ah I see - I gather it's meant to signal that all the test actions have been completed. This makes sense to me now.

Reply:
Yeah.  So previously we were just waiting for 16s (the default is to flush events every 15s) and then hoping they'd be there.  The sentinel event is the _last_ event that gets sent on any page, and signals that all of the previous ones should now be available.  The test polls for that event instead of just waiting, and then moves ahead when it sees it.  This cuts the wait times significantly.

Reply:
Got it! Yup definitely sounds like an improvement.